src/main/java/org/tugraz/sysds/runtime/instructions/spark/data/PartitionedBroadcast.java

```diff
@@ -96,7 +103,27 @@ public T getBlock(int rowIndex, int colIndex) {
 		
 		return _pbc[pix].value().getBlock(rowIndex, colIndex);
 	}
-	
+
+	public T getBlock(int[] ix) {
+		int pix = 0;
+		if( _pbc.length > 1 ) { //compute partition index
+			long[] dims = new long[_dc.getNumDims()];
+			int[] blen = new int[_dc.getNumDims()];
+			for (int i = 0; i < dims.length; i++) {
+				dims[i] = _dc.getDim(i);
+				blen[i] = (int) _dc.getBlockSize(i);
+			}
+			int numPerPart = computeBlocksPerPartition(dims, blen);
+			long l = blen[blen.length - 1];
+			for (int i = blen.length - 2; i >= 0; i--) {
+				l += (ix[i] - 1) * blen[i + 1];
+			}
```

Comment:
Please make this duplicated index computation a util method somewhere - maybe in `UtilFunctions`.

Reply:
Are you referring to the calculation of `long l`? Or to the part of getting `dims` and `blen` (which I changed now)?